wled00/FXparticleSystem.cpp

```
@@ -0,0 +1,1925 @@
+
+// apply a force in x,y direction to all particles
+// force is in 3.4 fixed point notation (see above)
+void ParticleSystem2D::applyForce(const int8_t xforce, const int8_t yforce) {
+  // for small forces, need to use a delay counter
+  uint8_t tempcounter;
+  // note: this is not the most computationally efficient way to do this, but it saves on duplicate code and is fast enough
+  for (uint32_t i = 0; i < usedParticles; i++) {
+    tempcounter = forcecounter;
+    applyForce(particles[i], xforce, yforce, tempcounter);
+  }
+  forcecounter = tempcounter; // save value back
```

Cppcheck (2.10-2)

[warning] Uninitialized variable

(uninitvar)